src/components/SignMessage.vue

```diff
@@ -17,43 +16,104 @@ const route = useRoute()
 
 defineProps({
   request: {
-    type: Request,
+    type: Object,
     required: true,
   },
 })
 
 const emits = defineEmits(['reject', 'approve'])
+const userStore = useUserStore()
 
-const stateChangeRequests = [
-  methodAndAction.wallet_addEthereumChain,
-  methodAndAction.wallet_switchEthereumChain,
-  methodAndAction.wallet_watchAsset,
-]
+function isSiweMessage(message: string) {
+  return (
+    message.includes('wants you to sign in with your Ethereum account') &&
+    message.includes('URI') &&
+    message.includes('Nonce') &&
+    message.includes('Version') &&
+    message.includes('Issued At') &&
+    message.toLowerCase().includes(userStore.walletAddress.toLowerCase())
+  )
+}
 
-function getTitle(requestMethod: string) {
-  if (stateChangeRequests.includes(requestMethod)) {
-    return requestMethod
+function getPermissionText(method, request) {
+  const { params } = request
+  let param: any
+  if (params instanceof Array && params[0]) {
+    param = params[0]
+  }
+  let response = 'performing an action'
+  switch (method) {
+    case 'wallet_addEthereumChain':
+      response = param?.chainName
+        ? `adding chain - ${param?.chainName}`
+        : 'adding a chain'
+      break
+    case 'wallet_switchEthereumChain':
+      response = param?.chainName
+        ? `switching chain - ${param?.chainName}`
+        : 'switching a chain'
+      break
+    case 'wallet_watchAsset':
+      response = 'adding a token'
+      break
+    case 'eth_sendTransaction':
+      response = 'sending a transaction'
+      break
+    case 'personal_sign':
+      if (isSiweMessage(param)) {
+        response = 'log in'
+      } else {
+        response = 'signing a message'
+      }
+      break
+    case 'eth_sign':
+      response = 'signing a message'
+      break
+    case 'eth_signTypedData_v4':
+      response = 'signing typed data'
+      break
+    case 'eth_decrypt':
+      response = 'decrypting data'
+      break
+    case 'eth_signTransaction':
+      response = 'signing a transaction'
+      break
+    default:
+      response = 'performing an action'
+      break
   }
-  return 'Sign Message'
+  return response
+}
+
+function isDeprecatedMethod(method) {
+  return method === 'eth_sign'
```

Comment:
can we use constant var here

like const ETH_SIGN = 'eth_sign' and use it here.

Reply:
Updated https://github.com/arcana-network/wallet-ui/pull/481/commits/a2ee76523718462443223abcafeaecf561688b41